content/bn/client-server-architecture.md

```diff
@@ -0,0 +1,19 @@
+---
+title: ক্লায়েন্ট-সার্ভার আর্কিটেকচার
+status: Feedback Appreciated
+category: প্রযুক্তি
+---
+
+## এটা কি
+
+একটি ক্লায়েন্ট-সার্ভার আর্কিটেকচারে (Client-Server Architecture), একটি অ্যাপ্লিকেশন তৈরি করে এমন যুক্তি (Code) দুটি বা ততোধিক উপাদানের মধ্যে বিভক্ত হয়: একটি ক্লায়েন্ট যা কাজ করতে বলে (যেমন আপনার ওয়েব ব্রাউজারে চলমান Gmail ওয়েব অ্যাপ্লিকেশন), এবং একটি বা আরও বেশি সার্ভার যা সেই অনুরোধটি পূরণ করে (যেমন, ক্লাউডে Google-এর কম্পিউটারে চলমান “ইমেল পাঠান”(Send Email) পরিষেবা)। এই উদাহরণে, আপনি যে বহির্গামী ইমেলগুলি লেখেন তা ক্লায়েন্ট দ্বারা পাঠানো হয় (আপনার ওয়েব ব্রাউজারে চলছে ওয়েব অ্যাপ্লিকেশন) একটি সার্ভারে (Gmail এর কম্পিউটার, যা আপনার বহির্গামী ইমেলগুলি তাদের প্রাপকদের কাছে ফরোয়ার্ড করে)।
+
+এটি স্বয়ংসম্পূর্ণ অ্যাপ্লিকেশনগুলির সাথে বৈপরীত্য (যেমন ডেস্কটপ অ্যাপ্লিকেশন) যা সমস্ত কাজ এক জায়গায় করে। উদাহরণস্বরূপ, মাইক্রোসফ্ট ওয়ার্ডের মতো একটি ওয়ার্ড প্রসেসিং প্রোগ্রাম install করা এবং সম্পূর্ণরূপে আপনার কম্পিউটারে চালানো হতে পারে।
+
+## এটা যেসব সমস্যাতে দৃষ্টিপাত করে
+
+একটি ক্লায়েন্ট-সার্ভার আর্কিটেকচার একটি বড় চ্যালেঞ্জ স্বয়ংসম্পূর্ণ অ্যাপ্লিকেশন পোজ সমাধান করে: নিয়মিত আপডেট। একটি স্বয়ংসম্পূর্ণ অ্যাপে, প্রতিটি আপডেটের জন্য, ব্যবহারকারীদের সর্বশেষ সংস্করণ ডাউনলোড এবং install করতে হবে। এটি ব্রাউজ করতে সক্ষম হওয়ার আগে আপনার নিজের কম্পিউটারে Amazon এর সমস্ত পণ্য ক্যাটালগ ডাউনলোড করার কথা ভাবুন!
+
+## এটা কিভাবে সাহায্য করে
+
+একটি দূরবর্তী সার্ভার বা পরিষেবাতে অ্যাপ্লিকেশন লজিক প্রয়োগ করে, অপারেটররা ক্লায়েন্ট-সাইডে যুক্তি পরিবর্তন করার প্রয়োজন ছাড়াই এটি আপডেট করতে পারে। এর মানে আপডেট অনেক বেশি ঘন ঘন করা যেতে পারে। সার্ভারে ডেটা সঞ্চয় করা অনেক ক্লায়েন্টকে একই ডেটা দেখতে এবং শেয়ার করতে দেয়। প্রথাগত অফলাইন ওয়ার্ড প্রসেসরের তুলনায় একটি অনলাইন ওয়ার্ড প্রসেসর ব্যবহারের মধ্যে পার্থক্য বিবেচনা করুন। পূর্বে, আপনার ফাইলগুলি সার্ভার-সাইডে বিদ্যমান এবং অন্যান্য ব্যবহারকারীদের সাথে ভাগ করা যেতে পারে যারা কেবল সার্ভার থেকে ডাউনলোড করে। লিগ্যাসি ওয়ার্ল্ডে, ফাইলগুলিকে অপসারণযোগ্য মিডিয়া (ফ্লপি ডিস্ক!,Floppy Disk) তে কপি করা এবং ব্যক্তিদের সাথে ভাগ করা প্রয়োজন৷
```

Comment:
It would be great if you could rearrange this sentence to a good one I guess "একটি দূরবর্তী সার্ভার বা পরিষেবাতে অ্যাপ্লিকেশন লজিক প্রয়োগ করে, অপারেটররা ক্লায়েন্ট-সাইডে যুক্তি পরিবর্তন করার প্রয়োজন ছাড়াই এটি আপডেট করতে পারে।"